### PR TITLE
Added support for providing utf flag to xcpretty via gym/xcodebuild

### PR DESCRIPTION
**Pull request by ghost**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [✔︎] Run `bundle exec rspec` from the root directory.
- [✔︎] Run `bundle exec rubocop -a` to ensure the code style is valid
- [✔︎] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [✔︎] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

This change adds support for passing the 'utf' flag to xcpretty, which is useful when you don't have strict control over your bash environment to apply the usual encoding fix in .profile.

### CODE REVIEW

File: fastlane/lib/fastlane/actions/xcodebuild.rb [hunk 1]
```diff
@@ -108,6 +111,10 @@ def self.run(params)
             raw_buildlog = params[:raw_buildlog]
           end
 
+          if params.key? :xcpretty_utf
+            xcpretty_utf = params[:xcpretty_utf]
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/xcodebuild.rb]: Can't we just combine that with https://github.com/fastlane/fastlane/pull/7659/files#diff-ce8c49563f8022fccd8ec5739763a461R102, and have it in one line?

File: fastlane/lib/fastlane/actions/xcodebuild.rb [hunk 2]
```diff
@@ -248,6 +255,7 @@ def self.run(params)
 
         xcpretty_command = ""
         xcpretty_command = "| xcpretty #{xcpretty_args}" unless raw_buildlog
+        xcpretty_command = "#{xcpretty_command} --utf" unless raw_buildlog || !xcpretty_utf
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/xcodebuild.rb]: unless in combination with || and ! is pretty confusing, can you rewrite that line so it's easier to read?

**Comment by lyptt** [comment]
I simplified the `unless` statement by splitting it out into two parts so it's easier to reason about what it's doing. I wasn't 100% sure what you meant regarding combining statements, it was hard to tell from the highlighted line alone, however I updated that particular statement to be one line to simplify things a little.

File: fastlane/lib/fastlane/actions/xcodebuild.rb [hunk 3]
```diff
@@ -417,7 +425,8 @@ def self.available_options
           ['output_style', 'Set the output format to one of: :standard (Colored UTF8 output, default), :basic (black & white ASCII output)'],
           ['buildlog_path', 'The path where the xcodebuild.log will be created, by default it is created in ~/Library/Logs/fastlane/xcbuild'],
           ['raw_buildlog', 'Set to true to see xcodebuild raw output. Default value is false'],
-          ['xcpretty_output', 'specifies the output type for xcpretty. eg. \'test\', or \'simple\'']
+          ['xcpretty_output', 'specifies the output type for xcpretty. eg. \'test\', or \'simple\''],
+          ['xcpretty_utf', 'Specifies xcpretty should use utf8 when reporting builds. This has no effect when raw_buildlog is specif    ied.']
```
Comment (mgrebenets) [inline fastlane/lib/fastlane/actions/xcodebuild.rb]: some extra spaces in `specified`?

File: fastlane/lib/fastlane/actions/xcodebuild.rb [hunk 4]
```diff
@@ -462,7 +471,8 @@ def self.available_options
...
```
Comment (mgrebenets) [inline fastlane/lib/fastlane/actions/xcodebuild.rb]: Same here,  seems like formatting issue.

File: fastlane/lib/fastlane/actions/xcodebuild.rb [hunk 5]
```diff
@@ -507,7 +517,8 @@ def self.available_options
...
```
Comment (mgrebenets) [inline fastlane/lib/fastlane/actions/xcodebuild.rb]: Another one with extra spaces in the "specified"

File: fastlane/lib/fastlane/actions/xcodebuild.rb [hunk 6]
```diff
@@ -548,7 +559,8 @@ def self.available_options
           ['output_style', 'Set the output format to one of: :standard (Colored UTF8 output, default), :basic (black & white ASCII output)'],
           ['buildlog_path', 'The path where the xcodebuild.log will be created, by default it is created in ~/Library/Logs/fastlane/xcbuild'],
           ['raw_buildlog', 'Set to true to see xcodebuild raw output. Default value is false'],
-          ['xcpretty_output', 'specifies the output type for xcpretty. eg. \'test\', or \'simple\'']
+          ['xcpretty_output', 'specifies the output type for xcpretty. eg. \'test\', or \'simple\''],
```
Comment (mgrebenets) [inline fastlane/lib/fastlane/actions/xcodebuild.rb]: ditto "specified"

File: fastlane/lib/fastlane/actions/xcodebuild.rb [hunk 7]
```diff
@@ -597,7 +609,8 @@ def self.available_options
...
```
Comment (mgrebenets) [inline fastlane/lib/fastlane/actions/xcodebuild.rb]: ditto "specified"

**Comment by mfurtak** [comment]
Added a quick commit to remove the extra spaces 👍

**Comment by mfurtak** [comment]
Hm, this may need to be rebased onto master to pick up the `rubocop` fixes.

**Comment by mfurtak** [comment]
Carried forward in #7891

**Pull request closed without merging**